File: Python/DesignPatterns/Builder.py
# Builder Demo in Python

class SimpleHtmlBuilder(object):
    def __init__(self):
        self.bodyContent = ''
    
    # Basic methods:
    def AddH1(self, text):
        self.bodyContent += '<h1>' + text + '</h1>'
    
    def AddH2(self, text):
        self.bodyContent += '<h2>' + text + '</h2>'

    def AddParagraph(self, text):
        self.bodyContent += '<p>' + text + '</p>'

    def AddHorizontalRule(self):
        self.bodyContent += '<hr>'

    def AddLineBreak(self):
        self.bodyContent += '<br>'

    # Builds the html document based on our input:
    def BuildDocument(self):
        # We'll follow the HTML 5 standard document syntax:
        return '''
        <!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Document</title>
</head>
<body>
    {}
</body>
</html>'''.format(self.bodyContent)



# Demo of using our simple html builder:

htmlBuilder = SimpleHtmlBuilder()

# Create the content of the page:
htmlBuilder.AddH1("Welcome to my blog!")
htmlBuilder.AddHorizontalRule()
htmlBuilder.AddH2("What is my blog about?")
htmlBuilder.AddParagraph("Good question! I'm not so sure yet, but you can't rush creativity.")
htmlBuilder.AddLineBreak()
htmlBuilder.AddHorizontalRule()

# Build the html:
print(htmlBuilder.BuildDocument())

File: PHP/DesignPatterns/Builder.php
<?php

class SimpleHtmlBuilder
{
    // The HTML content we'll display:
    private $bodyContent;

    function __construct()
    {
        $this->bodyContent = "";
    }

    public function AddH1($text)
    {
        $this->bodyContent .= "<h1>{$text}</h1>";
    }

    public function AddH2($text)
    {
        $this->bodyContent .= "<h2>{$text}</h2>";
    }

    public function AddParagraph($text)
    {
        $this->bodyContent .= "<p>{$text}</p>";
    }

    public function AddHorizontalRule()
    {
        $this->bodyContent .= "<hr>";
    }

    public function AddLineBreak()
    {
        $this->bodyContent .= "<br>";
    }

    // Creates the HTML document content based on HTML5 standards.
    public function BuildDocument()
    {
        return "<!DOCTYPE html>
        <html lang=\"en\">
        <head>
            <meta charset=\"UTF-8\">
            <meta name=\"viewport\" content=\"width=device-width, initial-scale=1.0\">
            <title>Document</title>
        </head>
        <body>
            {$this->bodyContent}
        </body>
        </html>";
    }
}